application/classes/Ushahidi/Console/PostExporter.php

```diff
@@ -141,23 +141,47 @@ protected function executeExport(InputInterface $input, OutputInterface $output)
         foreach ($filters as $key => $filter) {
             $data->$key = $filter;
 		}
+		$this->postExportRepository->setSearchParams($data);
+		$posts = $this->postExportRepository->getSearchResults();
+		service("formatter.entity.post.$format")->setFileSystem($this->fs);
+		service("formatter.entity.post.$format")->setAddHeader($add_header);
 
-        $this->postExportRepository->setSearchParams($data);
-		
-        $posts = $this->postExportRepository->getSearchResults();
+		$form_ids = $this->postExportRepository->getFormIdsForHeaders();
+		// we still need thenative fields such as create date
+		$attributes = $this->postExportRepository->getAttributes($form_ids);
+		$attributes[] = ['label' => 'Post ID', 'key' => 'id', 'type' => 'integer', 'input' => 'number', 'form_id' => 0, 'form_stage_id' => 0, 'form_stage_priority' => 0, 'priority' => 1];
+		$attributes[] = ['label' => 'Created (UTC)','key' => 'created', 'type' => 'datetime', 'input' => 'native', 'form_id' => 0, 'form_stage_id' => 0, 'form_stage_priority' => 0, 'priority' => 2];
+		$attributes[] = ['label' => 'Updated (UTC)', 'key' => 'updated', 'type' => 'datetime', 'input' => 'native', 'form_id' => 0, 'form_stage_id' => 0, 'form_stage_priority' => 0, 'priority' => 3];
+		$attributes[] = ['label' => 'Contact ID','key' => 'contact_id', 'type' => 'integer', 'input' => 'number', 'form_id' => 0, 'form_stage_id' => 0, 'form_stage_priority' => 0, 'priority' => 4];
+		$attributes[] = ['label' => 'Contact', 'key' => 'contact', 'type' => 'text', 'input' => 'text', 'form_id' => 0, 'form_stage_id' => 0, 'form_stage_priority' => 0, 'priority' => 5];
+		$keyAttributes = [];
+		foreach($attributes as $key => $item)
+		{
+			$keyAttributes[$item['key']] = $item;
+		}
 
 		// // ... remove any entities that cannot be seen
 		foreach ($posts as $idx => $post) {
-
 			// Retrieved Attribute Labels for Entity's values
-			$post = $this->postExportRepository->retrieveColumnNameData($post->asArray());
-
-			$posts[$idx] = $post;
+			$post = $this->postExportRepository->retrieveMetaData($post->asArray(), $keyAttributes);
+ 			$posts[$idx] = $post;
+ 		}
+		/**FIXME: how to make sure header_row is null/empty instead off an array with an empty item in it? */
+		if (empty($job->header_row) || $job->header_row[0] == '') {
+
+			$header_row = service("formatter.entity.post.$format")->createHeading($attributes, $posts);
+			$job->setState(['header_row' => $header_row]);
+			//FIXME update does not exist $job->update();
```

Comment:
oh @willdoran $job says update does not exist.

Reply:
That's weird it should because it inherits from Repository, I'll look